static/js/common/upload-addon.js

```diff
@@ -343,7 +343,9 @@
                               $("<p>").text(gettext("Your submission passed validation and will go through a manual review.")).appendTo(upload_results);
                             }
                           } else {
-                            $("<p>").text(gettext("Your submission didn't pass automatic validation and will go through a manual review.")).appendTo(upload_results);
+                            // If unlisted and not sideload and failed validation, disable submit until checkbox checked.
+                            $('.addon-upload-dependant').attr('disabled', true);
+                            $('#manual-review').show().removeClass('hidden');
```

Comment:
As explained previously, I think it needs to uncheck the checkbox here, just in case it was previously checked.